rbac/management/relation_replicator/outbox_replicator.py

```diff
@@ -39,33 +50,52 @@ def __init__(self, log: Optional["OutboxLog"] = None):
     def replicate(self, event: ReplicationEvent):
         """Replicate the given event to Kessel Relations via the Outbox."""
         payload = self._build_replication_event(event.add, event.remove)
-        self._save_replication_event(payload, event.event_type, event.event_info, event.partition_key)
+        self._save_replication_event(payload, event.event_type, event.event_info, str(event.partition_key))
 
-    def _build_replication_event(self, relations_to_add, relations_to_remove):
+    def _build_replication_event(
+        self, relations_to_add: list[Relationship], relations_to_remove: list[Relationship]
+    ) -> ReplicationEventPayload:
         """Build replication event."""
-        add_json = []
+        add_json: list[dict[str, Any]] = []
         for relation in relations_to_add:
             add_json.append(json_format.MessageToDict(relation))
 
-        remove_json = []
+        remove_json: list[dict[str, Any]] = []
         for relation in relations_to_remove:
             remove_json.append(json_format.MessageToDict(relation))
 
-        replication_event = {"relations_to_add": add_json, "relations_to_remove": remove_json}
-        return replication_event
+        return {"relations_to_add": add_json, "relations_to_remove": remove_json}
 
-    def _save_replication_event(self, payload, event_type, event_info: dict[str, object], aggregateid):
+    def _save_replication_event(
+        self,
+        payload: ReplicationEventPayload,
+        event_type: ReplicationEventType,
+        event_info: dict[str, object],
+        aggregateid: str,
+    ):
         """Save replication event."""
         # TODO: Can we add these as proper fields for kibana but also get logged in simple formatter?
+        logged_info = " ".join([f"info.{key}='{str(value)}'" for key, value in event_info.items()])
+
+        if not payload["relations_to_add"] and not payload["relations_to_remove"]:
```

Comment:
Does this work only when `payload["relations_to_add"] is None` or should it consider also empty array(s) ?

Reply:
Good question! It should also work with either I think, and I think `None` would be a type error. The tests use empty arrays.

Reply:
That's really cool!